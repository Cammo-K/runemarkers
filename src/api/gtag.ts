export enum EventNames {
  copyTileMarkers = `copy_tile_markers`,
}

type EventParams = {
  event_category?: string;
  event_label?: string;
  value?: number;
};

type Event = {
  action: EventNames;
  params: EventParams;
};

export const event = (event: Event) => {
  if (typeof window.gtag === `function`) {
    window.gtag(`event`, event.action, event.params);
  }
};
