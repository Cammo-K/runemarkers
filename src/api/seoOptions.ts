import { NextSeoProps } from 'next-seo';

const nextSeoOptions: NextSeoProps = {
  defaultTitle: `RuneMarkers - A Collection of RuneLite Ground Markers`,
  description: `A collection of ground (tile) markers for RuneLite. Find and import ground (tile) markers for different Oldschool RuneScape activities.`,
  titleTemplate: `RuneMarkers - %s`,
  canonical: `https://runemarkers.net`,
  openGraph: {
    type: `website`,
    locale: `en_US`,
    url: `https://runemarkers.net`,
    title: `RuneMarkers - A Collection of RuneLite Ground Markers`,
    description: `A collection of ground markers for RuneLite. Find and import ground markers for different Oldschool RuneScape activities.`,
    site_name: `RuneMarkers`,
  },
  twitter: {
    handle: `@jamgyo`,
    site: `@runemarkers`,
    cardType: `summary_large_image`,
  },
};

export const defaultImages = [
  {
    url: `/logo1024-background.png`,
    width: 1024,
    height: 1024,
    alt: `RuneMarkers - A Collection of RuneLite Ground Markers`,
  },
  {
    url: `/logo512-background.png`,
    width: 512,
    height: 512,
    alt: `RuneMarkers - A Collection of RuneLite Ground Markers`,
  },
  {
    url: `/android-chrome-512x512.png`,
    width: 512,
    height: 512,
    alt: `RuneMarkers - A Collection of RuneLite Ground Markers`,
  },
  {
    url: `/android-chrome-192x192.png`,
    width: 192,
    height: 192,
    alt: `RuneMarkers - A Collection of RuneLite Ground Markers`,
  },
];

export default nextSeoOptions;
