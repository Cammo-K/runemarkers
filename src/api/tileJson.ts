import tileJson from '@/tiles.json';
import { TileEntity } from '@/types';

export const tileData: TileEntity[] = tileJson.map((e) => ({
  safeURI: e.name.replaceAll(`:`, `-`),
  ...e,
}));
