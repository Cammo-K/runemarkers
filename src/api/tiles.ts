import { TileEntity } from '@/types';
import fs from 'fs';

export const getTileData: () => TileEntity[] = () =>
  fs
    .readdirSync(`./src/tiles`, {
      encoding: `utf-8`,
      withFileTypes: true,
    })
    .filter((file) => file.isFile() && file.name.endsWith(`.json`))
    .map(
      (file) =>
        ({
          safeURI: file.name.replace(`.json`, ``),
          ...JSON.parse(
            fs.readFileSync(`./src/tiles/${file.name}`, { encoding: `utf-8` }),
          ),
        } as TileEntity),
    );

//  = tileJson.map((e) => ({
//   safeURI: e.name.replaceAll(`:`, `-`),
//   ...e,
// }));
