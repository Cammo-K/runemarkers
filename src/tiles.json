[
  {
    "name": "Abyssal Sire",
    "thumbnail": "https://oldschool.runescape.wiki/images/thumb/Abyssal_Sire.png/200px-Abyssal_Sire.png",
    "wiki": "https://oldschool.runescape.wiki/w/Abyssal_Sire",
    "tags": [
      "slayer",
      "boss"
    ],
    "recommendedGuideVideoId": "wnZJl9driUs",
    "source": "https://www.youtube.com/watch?v=wnZJl9driUs",
    "tiles": [
      {
        "regionId": 11850,
        "regionX": 27,
        "regionY": 34,
        "z": 0,
        "color": "#FFFAFF00"
      },
      {
        "regionId": 11850,
        "regionX": 25,
        "regionY": 34,
        "z": 0,
        "color": "#FFFAFF00"
      },
      {
        "regionId": 11850,
        "regionX": 25,
        "regionY": 36,
        "z": 0,
        "color": "#FFFAFF00"
      },
      {
        "regionId": 11850,
        "regionX": 27,
        "regionY": 36,
        "z": 0,
        "color": "#FFFAFF00"
      },
      {
        "regionId": 11850,
        "regionX": 27,
        "regionY": 44,
        "z": 0,
        "color": "#FFFAFF00"
      },
      {
        "regionId": 11850,
        "regionX": 25,
        "regionY": 44,
        "z": 0,
        "color": "#FFFAFF00"
      }
    ]
  },
  {
    "name": "Alchemical Hydra",
    "thumbnail": "https://oldschool.runescape.wiki/images/thumb/Alchemical_Hydra.png/200px-Alchemical_Hydra.png",
    "wiki": "https://oldschool.runescape.wiki/w/Alchemical_Hydra",
    "tags": [
      "slayer",
      "boss"
    ],
    "recommendedGuideVideoId": "7ehTfsD--gM",
    "source": "https://www.youtube.com/watch?v=7ehTfsD--gM",
    "tiles": [
      {
        "regionId": 5536,
        "regionX": 27,
        "regionY": 38,
        "z": 0,
        "color": "#FFFFFF00"
      },
      {
        "regionId": 5536,
        "regionX": 20,
        "regionY": 31,
        "z": 0,
        "color": "#FFFFFF00"
      },
      {
        "regionId": 5536,
        "regionX": 19,
        "regionY": 30,
        "z": 0,
        "color": "#FFFFFF00"
      },
      {
        "regionId": 5536,
        "regionX": 33,
        "regionY": 23,
        "z": 0,
        "color": "#FFFFFF00"
      },
      {
        "regionId": 5536,
        "regionX": 14,
        "regionY": 38,
        "z": 0,
        "color": "#FFFFFF00"
      },
      {
        "regionId": 5536,
        "regionX": 13,
        "regionY": 32,
        "z": 0,
        "color": "#FFFFFF00"
      },
      {
        "regionId": 5536,
        "regionX": 18,
        "regionY": 30,
        "z": 0,
        "color": "#FF747474"
      }
    ]
  },
  {
    "name": "Commander Zilyana",
    "altName": "Saradomin",
    "thumbnail": "https://oldschool.runescape.wiki/images/thumb/Commander_Zilyana.png/200px-Commander_Zilyana.png",
    "wiki": "https://oldschool.runescape.wiki/w/Commander_Zilyana",
    "tags": [
      "boss",
      "godwars",
      "gwd",
      "saradomin"
    ],
    "recommendedGuideVideoId": "-2WPwh2NRIE",
    "source": "https://www.youtube.com/watch?v=-2WPwh2NRIE",
    "tiles": [
      {
        "regionId": 11602,
        "regionX": 27,
        "regionY": 10,
        "z": 0,
        "color": "#FFFAFF00"
      },
      {
        "regionId": 11602,
        "regionX": 27,
        "regionY": 27,
        "z": 0,
        "color": "#FFFAFF00"
      },
      {
        "regionId": 11602,
        "regionX": 9,
        "regionY": 27,
        "z": 0,
        "color": "#FFFAFF00"
      },
      {
        "regionId": 11602,
        "regionX": 9,
        "regionY": 10,
        "z": 0,
        "color": "#FFFAFF00"
      },
      {
        "regionId": 11602,
        "regionX": 26,
        "regionY": 24,
        "z": 0,
        "color": "#FFFAFF00"
      }
    ]
  },
  {
    "name": "Commander Zilyana (No Stam)",
    "altName": "Saradomin (No Stam)",
    "thumbnail": "https://oldschool.runescape.wiki/images/thumb/Commander_Zilyana.png/200px-Commander_Zilyana.png",
    "wiki": "https://oldschool.runescape.wiki/w/Commander_Zilyana",
    "tags": [
      "boss",
      "godwars",
      "gwd"
    ],
    "source": "https://www.youtube.com/watch?v=WUa4T1lRNKo",
    "recommendedGuideVideoId": "WUa4T1lRNKo",
    "tiles": [
      {
        "regionId": 11602,
        "regionX": 26,
        "regionY": 12,
        "z": 0,
        "color": "#63FFFF00"
      },
      {
        "regionId": 11602,
        "regionX": 19,
        "regionY": 10,
        "z": 0,
        "color": "#63FFFF00"
      },
      {
        "regionId": 11602,
        "regionX": 18,
        "regionY": 10,
        "z": 0,
        "color": "#63FFFF00"
      },
      {
        "regionId": 11602,
        "regionX": 14,
        "regionY": 10,
        "z": 0,
        "color": "#63FFFF00"
      },
      {
        "regionId": 11602,
        "regionX": 9,
        "regionY": 19,
        "z": 0,
        "color": "#63FFFF00"
      },
      {
        "regionId": 11602,
        "regionX": 9,
        "regionY": 20,
        "z": 0,
        "color": "#63FFFF00"
      },
      {
        "regionId": 11602,
        "regionX": 18,
        "regionY": 25,
        "z": 0,
        "color": "#63FFFF00"
      },
      {
        "regionId": 11602,
        "regionX": 19,
        "regionY": 25,
        "z": 0,
        "color": "#63FFFF00"
      },
      {
        "regionId": 11602,
        "regionX": 23,
        "regionY": 25,
        "z": 0,
        "color": "#63FFFF00"
      },
      {
        "regionId": 11602,
        "regionX": 24,
        "regionY": 25,
        "z": 0,
        "color": "#63FFFF00"
      },
      {
        "regionId": 11602,
        "regionX": 26,
        "regionY": 17,
        "z": 0,
        "color": "#63FFFF00"
      },
      {
        "regionId": 11602,
        "regionX": 26,
        "regionY": 16,
        "z": 0,
        "color": "#63FFFF00"
      },
      {
        "regionId": 11602,
        "regionX": 26,
        "regionY": 18,
        "z": 0,
        "color": "#63FF0000"
      },
      {
        "regionId": 11602,
        "regionX": 27,
        "regionY": 25,
        "z": 0,
        "color": "#63FF0000"
      },
      {
        "regionId": 11602,
        "regionX": 22,
        "regionY": 25,
        "z": 0,
        "color": "#63FF0000"
      },
      {
        "regionId": 11602,
        "regionX": 17,
        "regionY": 25,
        "z": 0,
        "color": "#63FF0000"
      },
      {
        "regionId": 11602,
        "regionX": 11,
        "regionY": 25,
        "z": 0,
        "color": "#63FFFF00"
      },
      {
        "regionId": 11602,
        "regionX": 9,
        "regionY": 23,
        "z": 0,
        "color": "#63FF0000"
      },
      {
        "regionId": 11602,
        "regionX": 9,
        "regionY": 18,
        "z": 0,
        "color": "#63FF0000"
      },
      {
        "regionId": 11602,
        "regionX": 9,
        "regionY": 10,
        "z": 0,
        "color": "#63FF0000"
      },
      {
        "regionId": 11602,
        "regionX": 20,
        "regionY": 10,
        "z": 0,
        "color": "#63FF0000"
      },
      {
        "regionId": 11602,
        "regionX": 26,
        "regionY": 13,
        "z": 0,
        "color": "#63FF0000"
      },
      {
        "regionId": 11602,
        "regionX": 17,
        "regionY": 11,
        "z": 0,
        "color": "#630000FF"
      },
      {
        "regionId": 11602,
        "regionX": 26,
        "regionY": 10,
        "z": 0,
        "color": "#6300FFFF"
      },
      {
        "regionId": 11602,
        "regionX": 27,
        "regionY": 12,
        "z": 0,
        "color": "#6300FFFF"
      },
      {
        "regionId": 11602,
        "regionX": 27,
        "regionY": 11,
        "z": 0,
        "color": "#63FF00FF"
      },
      {
        "regionId": 11602,
        "regionX": 15,
        "regionY": 10,
        "z": 0,
        "color": "#63FF0000"
      }
    ]
  },
  {
    "name": "General Graardor (6:0)",
    "altName": "Bandos (6:0)",
    "thumbnail": "https://oldschool.runescape.wiki/images/thumb/General_Graardor.png/200px-General_Graardor.png",
    "wiki": "https://oldschool.runescape.wiki/w/General_Graardor",
    "tags": [
      "boss",
      "godwars",
      "gwd",
      "bandos"
    ],
    "recommendedGuideVideoId": "oIWnXBqebTU",
    "source": "https://www.youtube.com/watch?v=oIWnXBqebTU",
    "tiles": [
      {
        "regionId": 11347,
        "regionX": 60,
        "regionY": 39,
        "z": 2,
        "color": "#FFF8EA09"
      },
      {
        "regionId": 11347,
        "regionX": 48,
        "regionY": 39,
        "z": 2,
        "color": "#FFF8EA09"
      },
      {
        "regionId": 11347,
        "regionX": 48,
        "regionY": 57,
        "z": 2,
        "color": "#FFF8EA09"
      },
      {
        "regionId": 11347,
        "regionX": 60,
        "regionY": 54,
        "z": 2,
        "color": "#FFF8EA09"
      },
      {
        "regionId": 11347,
        "regionX": 48,
        "regionY": 49,
        "z": 2,
        "color": "#FFF8EA09"
      },
      {
        "regionId": 11347,
        "regionX": 60,
        "regionY": 46,
        "z": 2,
        "color": "#FFF8EA09"
      },
      {
        "regionId": 11347,
        "regionX": 57,
        "regionY": 40,
        "z": 2,
        "color": "#FFF8EA09",
        "label": "start"
      },
      {
        "regionId": 11347,
        "regionX": 50,
        "regionY": 48,
        "z": 2,
        "color": "#FFFF0000"
      },
      {
        "regionId": 11347,
        "regionX": 52,
        "regionY": 48,
        "z": 2,
        "color": "#FFFF0000"
      }
    ]
  },
  {
    "name": "General Graardor (9:0)",
    "altName": "Bandos (9:0)",
    "thumbnail": "https://oldschool.runescape.wiki/images/thumb/General_Graardor.png/200px-General_Graardor.png",
    "wiki": "https://oldschool.runescape.wiki/w/General_Graardor",
    "tags": [
      "boss",
      "godwars",
      "gwd",
      "bandos"
    ],
    "recommendedGuideVideoId": "YBKT-_USIQg",
    "source": "https://www.youtube.com/watch?v=YBKT-_USIQg",
    "tiles": [
      {
        "regionId": 11347,
        "regionX": 57,
        "regionY": 55,
        "z": 2,
        "color": "#00FF0000"
      },
      {
        "regionId": 11347,
        "regionX": 56,
        "regionY": 45,
        "z": 2,
        "color": "#00FF0000"
      },
      {
        "regionId": 11347,
        "regionX": 59,
        "regionY": 50,
        "z": 2,
        "color": "#FFFF9000"
      },
      {
        "regionId": 11347,
        "regionX": 57,
        "regionY": 57,
        "z": 2,
        "color": "#FF19A800",
        "label": "Start"
      },
      {
        "regionId": 11347,
        "regionX": 51,
        "regionY": 54,
        "z": 2,
        "color": "#FFFF0000",
        "label": "1"
      },
      {
        "regionId": 11347,
        "regionX": 55,
        "regionY": 48,
        "z": 2,
        "color": "#FFFF0000",
        "label": "2"
      },
      {
        "regionId": 11347,
        "regionX": 57,
        "regionY": 44,
        "z": 2,
        "color": "#FFFF0000",
        "label": "3"
      },
      {
        "regionId": 11347,
        "regionX": 51,
        "regionY": 48,
        "z": 2,
        "color": "#FFFF0000",
        "label": "4"
      },
      {
        "regionId": 11347,
        "regionX": 54,
        "regionY": 42,
        "z": 2,
        "color": "#FFFF0000",
        "label": "5"
      },
      {
        "regionId": 11347,
        "regionX": 49,
        "regionY": 39,
        "z": 2,
        "color": "#FFFF0000",
        "label": "6"
      },
      {
        "regionId": 11347,
        "regionX": 57,
        "regionY": 47,
        "z": 2,
        "color": "#FFFF0000",
        "label": "7"
      },
      {
        "regionId": 11347,
        "regionX": 57,
        "regionY": 52,
        "z": 2,
        "color": "#FFFF0000",
        "label": "8"
      },
      {
        "regionId": 11347,
        "regionX": 57,
        "regionY": 56,
        "z": 2,
        "color": "#FFFF0000",
        "label": "9"
      }
    ]
  },
  {
    "name": "Olm",
    "thumbnail": "https://oldschool.runescape.wiki/images/thumb/Great_Olm.png/200px-Great_Olm.png",
    "wiki": "https://oldschool.runescape.wiki/w/Great_Olm",
    "tags": [
      "cox",
      "chambers of xeric",
      "raids"
    ],
    "tiles": [
      {
        "regionId": 12889,
        "regionX": 37,
        "regionY": 50,
        "z": 0,
        "color": "#FFFAFF00"
      },
      {
        "regionId": 12889,
        "regionX": 37,
        "regionY": 38,
        "z": 0,
        "color": "#FFFAFF00"
      },
      {
        "regionId": 12889,
        "regionX": 28,
        "regionY": 38,
        "z": 0,
        "color": "#FFFAFF00"
      },
      {
        "regionId": 12889,
        "regionX": 28,
        "regionY": 50,
        "z": 0,
        "color": "#FFFAFF00"
      },
      {
        "regionId": 12889,
        "regionX": 28,
        "regionY": 47,
        "z": 0,
        "color": "#FFFAFF00",
        "label": "thumb"
      },
      {
        "regionId": 12889,
        "regionX": 37,
        "regionY": 41,
        "z": 0,
        "color": "#FFFAFF00",
        "label": "thumb"
      },
      {
        "regionId": 12889,
        "regionX": 37,
        "regionY": 45,
        "z": 0,
        "color": "#FFFAFF00"
      },
      {
        "regionId": 12889,
        "regionX": 37,
        "regionY": 43,
        "z": 0,
        "color": "#FFFAFF00"
      },
      {
        "regionId": 12889,
        "regionX": 28,
        "regionY": 43,
        "z": 0,
        "color": "#FFFAFF00"
      },
      {
        "regionId": 12889,
        "regionX": 28,
        "regionY": 45,
        "z": 0,
        "color": "#FFFAFF00"
      }
    ]
  },
  {
    "name": "Sarachnis",
    "thumbnail": "https://oldschool.runescape.wiki/images/thumb/Sarachnis.png/200px-Sarachnis.png",
    "wiki": "https://oldschool.runescape.wiki/w/Sarachnis",
    "tags": [
      "boss"
    ],
    "tiles": [
      {
        "regionId": 7322,
        "regionX": 47,
        "regionY": 43,
        "z": 0,
        "color": "#FFFAFF00"
      },
      {
        "regionId": 7322,
        "regionX": 47,
        "regionY": 48,
        "z": 0,
        "color": "#FFFAFF00"
      },
      {
        "regionId": 7322,
        "regionX": 52,
        "regionY": 48,
        "z": 0,
        "color": "#FFFAFF00"
      },
      {
        "regionId": 7322,
        "regionX": 52,
        "regionY": 43,
        "z": 0,
        "color": "#FFFAFF00"
      }
    ]
  },
  {
    "name": "Vorkath (melee)",
    "thumbnail": "https://oldschool.runescape.wiki/images/thumb/Vorkath.png/200px-Vorkath.png",
    "wiki": "https://oldschool.runescape.wiki/w/Vorkath",
    "tags": [
      "boss"
    ],
    "tiles": [
      {
        "regionId": 9023,
        "regionX": 32,
        "regionY": 29,
        "z": 0,
        "color": "#FF0090FF"
      },
      {
        "regionId": 9023,
        "regionX": 34,
        "regionY": 29,
        "z": 0,
        "color": "#FF0090FF"
      },
      {
        "regionId": 9023,
        "regionX": 30,
        "regionY": 29,
        "z": 0,
        "color": "#FF0090FF"
      },
      {
        "regionId": 9023,
        "regionX": 30,
        "regionY": 26,
        "z": 0,
        "color": "#FF0090FF"
      },
      {
        "regionId": 9023,
        "regionX": 31,
        "regionY": 26,
        "z": 0,
        "color": "#FF0090FF"
      },
      {
        "regionId": 9023,
        "regionX": 32,
        "regionY": 26,
        "z": 0,
        "color": "#FF0090FF"
      },
      {
        "regionId": 9023,
        "regionX": 33,
        "regionY": 26,
        "z": 0,
        "color": "#FF0090FF"
      },
      {
        "regionId": 9023,
        "regionX": 34,
        "regionY": 26,
        "z": 0,
        "color": "#FF0090FF"
      }
    ]
  }
]