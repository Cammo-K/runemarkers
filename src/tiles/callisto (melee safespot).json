{
  "name": "Callisto (Melee Safespot)",
  "thumbnail": "https://oldschool.runescape.wiki/images/thumb/Callisto.png/200px-Callisto.png",
  "wiki": "https://oldschool.runescape.wiki/w/Callisto",
  "tags": ["Wilderness", "Wildy"],
  "recommendGuideVideoId": "jcORL3xnPGE",
  "source": "https://www.youtube.com/watch?v=jcORL3xnPGE",
  "tiles": [
    {
      "regionId": 13371,
      "regionX": 0,
      "regionY": 61,
      "z": 0,
      "color": "#FFFAFF00",
      "label": "1"
    },
    {
      "regionId": 13371,
      "regionX": 5,
      "regionY": 61,
      "z": 0,
      "color": "#FFFAFF00",
      "label": "2"
    },
    {
      "regionId": 13371,
      "regionX": 9,
      "regionY": 33,
      "z": 0,
      "color": "#FFFAFF00",
      "label": "4"
    },
    {
      "regionId": 13371,
      "regionX": 6,
      "regionY": 32,
      "z": 0,
      "color": "#FFFAFF00",
      "label": "3"
    },
    {
      "regionId": 13371,
      "regionX": 8,
      "regionY": 41,
      "z": 0,
      "color": "#FFFAFF00",
      "label": "5"
    }
  ]
}
