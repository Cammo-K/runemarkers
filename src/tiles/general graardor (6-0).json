{
  "name": "General Graardor (6:0)",
  "altName": "Bandos (6:0)",
  "thumbnail": "https://oldschool.runescape.wiki/images/thumb/General_Graardor.png/200px-General_Graardor.png",
  "wiki": "https://oldschool.runescape.wiki/w/General_Graardor",
  "tags": [
    "boss",
    "godwars",
    "gwd",
    "bandos"
  ],
  "recommendedGuideVideoId": "oIWnXBqebTU",
  "source": "https://www.youtube.com/watch?v=oIWnXBqebTU",
  "tiles": [
    {
      "regionId": 11347,
      "regionX": 60,
      "regionY": 39,
      "z": 2,
      "color": "#FFF8EA09"
    },
    {
      "regionId": 11347,
      "regionX": 48,
      "regionY": 39,
      "z": 2,
      "color": "#FFF8EA09"
    },
    {
      "regionId": 11347,
      "regionX": 48,
      "regionY": 57,
      "z": 2,
      "color": "#FFF8EA09"
    },
    {
      "regionId": 11347,
      "regionX": 60,
      "regionY": 54,
      "z": 2,
      "color": "#FFF8EA09"
    },
    {
      "regionId": 11347,
      "regionX": 48,
      "regionY": 49,
      "z": 2,
      "color": "#FFF8EA09"
    },
    {
      "regionId": 11347,
      "regionX": 60,
      "regionY": 46,
      "z": 2,
      "color": "#FFF8EA09"
    },
    {
      "regionId": 11347,
      "regionX": 57,
      "regionY": 40,
      "z": 2,
      "color": "#FFF8EA09",
      "label": "start"
    },
    {
      "regionId": 11347,
      "regionX": 50,
      "regionY": 48,
      "z": 2,
      "color": "#FFFF0000"
    },
    {
      "regionId": 11347,
      "regionX": 52,
      "regionY": 48,
      "z": 2,
      "color": "#FFFF0000"
    }
  ]
}