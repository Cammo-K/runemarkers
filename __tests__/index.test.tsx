import { getTileData } from '@/api/tiles';
import '@testing-library/jest-dom';
import { act, fireEvent, render } from '@testing-library/react';
import Home, { getStaticProps } from '@/pages/index';

describe(`Home`, () => {
  it(`should search correctly`, () => {
    const data = getTileData();
    const { container, getByPlaceholderText } = render(
      <Home tileData={data} />,
    );
    const input = getByPlaceholderText(`type here to search`);
    act(() => {
      fireEvent.change(input, { target: { value: `abyssal sire` } });
    });
    expect(container).toMatchSnapshot();
  });

  it(`should generate static props`, async () => {
    const tileData = getTileData();
    const { props } = await getStaticProps();
    expect(props.tileData).toEqual(tileData);
  });
});
