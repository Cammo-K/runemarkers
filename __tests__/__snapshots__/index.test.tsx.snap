// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should search correctly 1`] = `
<div>
  <div>
    <main
      class="main"
    >
      <div
        class="navbar"
      >
        <a
          href="./"
          style="background: transparent; color: rgb(245, 243, 245); text-decoration: none; display: flex;"
        >
          <header
            class="title"
            style="font-size: 2rem;"
          >
            R
            <span>
              une
            </span>
            <span
              class="titleAlt"
            >
              M
              <span>
                arkers
              </span>
            </span>
          </header>
        </a>
      </div>
      <div
        class="container"
      >
        <p
          style="padding: 0px; color: rgb(199, 199, 199); text-align: center; min-width: 280px; margin-bottom: 15px; max-width: 600px; width: 80%;"
        >
          Make sure the "Ground Markers" plugin is enabled on RuneLite with "Remember color per tile" enabled. Copy the tile markers to your clipboard, right-click the world map in-game and select "import".
        </p>
        <div
          style="display: flex; align-items: center; margin-bottom: 20px;"
        >
          <input
            class="input"
            placeholder="type here to search"
            style="width: 175px; min-width: 50px; max-width: 500px;"
            type="text"
          />
        </div>
        <div
          class="innerContainer"
        >
          <div
            class="card"
          >
            <a
              class="imageContainer"
              href="/abyssal%20sire"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27140%27%20height=%27140%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="Abyssal Sire.png"
                  class="image"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </a>
            <div
              class="cardData"
            >
              <div
                class="entityInfo"
              >
                <a
                  class="name"
                  href="/abyssal%20sire"
                >
                  Abyssal Sire
                </a>
              </div>
              <div>
                <p
                  class="tileCount"
                >
                  6
                   tile markers
                </p>
                <div
                  class="tileInteraction"
                >
                  <button
                    class="button button"
                  >
                    Copy
                    <div
                      class="animBar "
                    />
                  </button>
                  <a
                    class="button"
                    href="/abyssal%20sire"
                  >
                    <svg
                      class="svg"
                      height="24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M11 17H13V11H11ZM12 9Q12.425 9 12.713 8.712Q13 8.425 13 8Q13 7.575 12.713 7.287Q12.425 7 12 7Q11.575 7 11.288 7.287Q11 7.575 11 8Q11 8.425 11.288 8.712Q11.575 9 12 9ZM12 22Q9.925 22 8.1 21.212Q6.275 20.425 4.925 19.075Q3.575 17.725 2.788 15.9Q2 14.075 2 12Q2 9.925 2.788 8.1Q3.575 6.275 4.925 4.925Q6.275 3.575 8.1 2.787Q9.925 2 12 2Q14.075 2 15.9 2.787Q17.725 3.575 19.075 4.925Q20.425 6.275 21.212 8.1Q22 9.925 22 12Q22 14.075 21.212 15.9Q20.425 17.725 19.075 19.075Q17.725 20.425 15.9 21.212Q14.075 22 12 22ZM12 12Q12 12 12 12Q12 12 12 12Q12 12 12 12Q12 12 12 12Q12 12 12 12Q12 12 12 12Q12 12 12 12Q12 12 12 12ZM12 20Q15.325 20 17.663 17.663Q20 15.325 20 12Q20 8.675 17.663 6.337Q15.325 4 12 4Q8.675 4 6.338 6.337Q4 8.675 4 12Q4 15.325 6.338 17.663Q8.675 20 12 20Z"
                      />
                    </svg>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <footer
        class="footer"
      >
        <div
          class="footerItem"
        >
          <p>
            Made by
            <a
              class="link"
              href="https://discord.com/users/167850724976361472"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2721%27%20height=%2724%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="GitHub"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
              <span>
                Jam#0001
              </span>
            </a>
          </p>
        </div>
        <div
          class="footerItem"
        >
          <a
            class="link"
            href="https://github.com/jamiegyoung/runemarkers"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="GitHub"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <span>
              contribute code / tiles
            </span>
          </a>
        </div>
      </footer>
    </main>
  </div>
</div>
`;
